golang/dep/dep-override/.gitignore

```diff
@@ -0,0 +1,2 @@
+example-go-golangdep
```

Comment:
This is for ignoring the compiled binary if any. In this case I think the binary will be named `dep-override` instead (it follows the directory's name). Please compile, check the binary's name, and update the value here. Thank you!

Reply:
@hendychua Good catch! I confirmed that the binary name is `dep-override` and updated `.gitignore` accordingly. I had also made `main.go` actually buildable.